### ISSUE TITLE
winml config/build for encoder-decoder models without --task emits a decoder-only, non-runnable model

### ISSUE THREAD

**Issue by timenick** [microsoft/winml-cli#850]
## Summary

For encoder-decoder model types (t5, marian, mu2, bart, vision-encoder-decoder, blip), running `winml config` / `build` **without `--task`** produces a **decoder-only** export — it takes `encoder_hidden_states` as an *input* but contains no encoder to produce it, so the result can't run standalone.

## Reproduction

```
winml config -m t5-small      # no --task
```

```
loader.task  : text2text-generation
loader.class : T5DecoderWrapper
inputs       : decoder_input_ids, encoder_hidden_states, past_0_key/value … past_5_*
outputs      : logits, present_0..5_key/value
```

The export is the decoder half only; `encoder_hidden_states` has no producer in the build.

## Root cause

The full encoder+decoder export is built via the **composite path**, which is keyed on concrete tasks (`(t5, summarization)`, `(t5, translation)`, `(bart, summarization)`, …) and only fires for an explicit `--task` — `_resolve_composite_model_components` returns early when `task is None`. With no `--task`, auto-detection resolves `text2text-generation`, which (a) is not a composite key and (b) maps to the single `*DecoderWrapper` in `MODEL_CLASS_MAPPING`, so only the decoder is exported.

## Scope

All encoder-decoder model types: t5, marian, mu2, vision-encoder-decoder, blip. (bart currently auto-detects `fill-mask` so it hits a *different* failure today; once the detection fix in #838 lands, bart joins the others in this decoder-only path.)

## Proposed fixes (pick one)

- **Route no-`--task` seq2seq through the full composite** — when auto-detection yields an encoder-decoder generation task for an `is_encoder_decoder` model, build the encoder+decoder composite instead of the decoder alone.
- **Fail loudly** — reject a no-`--task` build for encoder-decoder models with a clear message to pass `--task summarization|translation|…`, instead of silently emitting a decoder-only model.

**Comment by DingmaomaoBJTU** [comment]
## Triage Summary

**Priority:** P1
**Component:** config / build / loader
**Assigned:** @timenick

**Rationale:** Encoder-decoder models (t5, marian, bart, blip, etc.) without explicit --task produce decoder-only exports that cannot run standalone. This silently produces a broken artifact — users won't discover the issue until inference time. Impacts all encoder-decoder model families.

**Next steps:** When task is not specified for encoder-decoder architectures, either auto-detect the composite task or emit a clear error/warning that --task is required for these model types.
